pkg/search/postgres/store_cache.go

```diff
@@ -346,30 +316,21 @@ func (c *cachedStore[T, PT]) Walk(ctx context.Context, fn func(obj PT) error) er
 	return c.walkCacheNoLock(ctx, fn)
 }
 
+// GetByQueryFn iterates over the objects from the store matching the query.
+func (c *cachedStore[T, PT]) GetByQueryFn(ctx context.Context, query *v1.Query, fn func(obj PT) error) error {
+	defer c.setCacheOperationDurationTime(time.Now(), ops.GetByQuery)
+	return c.getByQueryFn(ctx, query, fn)
+}
+
 // GetByQuery returns the objects from the store matching the query.
 func (c *cachedStore[T, PT]) GetByQuery(ctx context.Context, query *v1.Query) ([]*T, error) {
 	defer c.setCacheOperationDurationTime(time.Now(), ops.GetByQuery)
-	identifiers, err := c.underlyingStore.GetIDsByQuery(ctx, query)
-	// Fallback to the underlying store on error.
-	if err != nil {
-		log.Errorf("Failed to get identifiers by query, falling back to get results by query: %v", err)
-		return c.underlyingStore.GetByQuery(ctx, query)
-	}
-	results := make([]*T, 0, len(identifiers))
-	c.cacheLock.RLock()
-	defer c.cacheLock.RUnlock()
-	for _, id := range identifiers {
-		obj, found := c.cache[id]
-		if !found {
-			log.Warnf("Object %q not found in store cache", id)
-			continue
-		}
-		if !c.isReadAllowed(ctx, obj) {
-			continue
-		}
-		results = append(results, obj.CloneVT())
-	}
-	return results, nil
+	results := make([]*T, 0, query.GetPagination().GetLimit())
```

Comment:
Should the limit be pre-computed to catch the IntMax value and not pre-allocated in that case ?

Reply:
how about using `batchAfter` instead?

Reply:
- https://github.com/stackrox/stackrox/pull/15129